payments/views.py

```diff
@@ -67,12 +67,15 @@ class WorkshopDetail(FormMixin, DetailView):
     form_class = OrderForm
     context_object_name = 'workshop'
 
+    # discount_code is needed on POST as well, so only getting it on
```

Comment:
I don't know if this `dispatch` is quite what we need. Query params on GET requests are generally frowned upon. How about we just stash the discount code in the session on GET? See below for more.